spec/truffle/undef_method_spec.rb

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 1.0, or
+# GNU General Public License version 2, or
+# GNU Lesser General Public License version 2.1.
+
+require_relative '../ruby/spec_helper'
+
+describe "Module#undef_method" do
+  describe "reports an error message as expected by minitest tests" do
+    class Object
+
+      def truffleruby_minitest_spec_stub
+        metaclass = class << self; self; end
+        metaclass.send :alias_method, :new_truffleruby_minitest_spec_stub_does_not_exist, :truffleruby_minitest_spec_stub_does_not_exist
+      ensure
+        metaclass.send :undef_method, :truffleruby_minitest_spec_stub_does_not_exist
+      end
+
+    end
+
+    -> { Time.truffleruby_minitest_spec_stub }.should raise_error(NameError, /undefined method `truffleruby_minitest_spec_stub_does_not_exist' for class `Time'/)
```

Comment:
Could you add specs in `spec/ruby/core/module/undef_method_spec.rb` instead?
Specs under `spec/truffle` should only contain things that cannot run on MRI or where our behavior differs intentionally.

The spec for the message can much simpler, as shown by my examples in the other comment.
Line 59 of `spec/ruby/core/module/undef_method_spec.rb` already has an example which can be copy-pasted and used to assert the message.
Please also add one for a `Class.new`, as the message differs there.

If there is special behavior for singleton classes, then please add a spec for that too.

Reply:
> Could you add specs in spec/ruby/core/module/undef_method_spec.rb instead?
Specs under spec/truffle should only contain things that cannot run on MRI or where our behavior differs intentionally.

Yes, as I already said, the spec is only there as I wasn't sure how to write it best for `spec/ruby` and I was asking for advice.